crates/proto/src/xfer/dns_handle.rs

```diff
@@ -41,6 +41,11 @@ pub trait DnsHandle: 'static + Clone + Send + Sync + Unpin {
         false
     }
 
+    /// Allow for disabling EDNS
+    fn is_using_edns(&self) -> bool {
```

Comment:
If I'm not mistaken, this is only used in `ClientHandle` contexts, can we instead define it on that?

Reply:
How? `ClientHandle` has a blanket implementation:
```rust
impl<T> ClientHandle for T where T: DnsHandle<Error = ProtoError> {}
```

Reply:
3 lines down is the trait implementation... I was suggesting moving the `is_using_edns` from `DnsHandle` to `ClientHandle`. If it's not too hard. That way we can keep this isolated in-case we want to change the way edns is treated in the future.

Reply:
The `impl<T> ClientHandle` can only access what `trait DnsHandle` provides, so ` is_using_edns()` would have to be defined in `DnsHandle` anyway.

Reply:
yes, you're correct.